From 82ecf9473c8203af24e0b5d5c72e161075d76cc9 Mon Sep 17 00:00:00 2001
From: "Schmidt, Andreas" <Andreas.Schmidt@wago.com>
Date: Tue, 27 Mar 2018 09:35:54 +0000
Subject: [PATCH 167/261] net:dsa: sleep after softreset for mv886321

One mv88e6321 in PFC200v3 8215 P50 hang while setup function.
Precise MDIO communication after softreset is not immediatly
usable. This patch add a sleep of 100ms after softreset of
mv88e6321.

Signed-off-by: Andreas Schmidt <andreas.schmidt@wago.com>
---
 drivers/net/dsa/mv88e6xxx/chip.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/dsa/mv88e6xxx/chip.c b/drivers/net/dsa/mv88e6xxx/chip.c
index 1c2649f..7c87e6b 100644
--- a/drivers/net/dsa/mv88e6xxx/chip.c
+++ b/drivers/net/dsa/mv88e6xxx/chip.c
@@ -2414,6 +2414,11 @@ static int mv88e6xxx_switch_reset(struct mv88e6xxx_chip *chip)
 	if (err)
 		return err;
 
+	/* Wait a while after soft reset. Some chips need about 100ms
+	 * they can call over mdio.
+	 */
+	msleep(150);
+
 	/* Wait up to one second for reset to complete. */
 	timeout = jiffies + 1 * HZ;
 	while (time_before(jiffies, timeout)) {
-- 
2.7.4

